docker-compose.jaeger.yml

```diff
@@ -0,0 +1,85 @@
+version: '3.7'
+
+services:
+  sia:
+    environment:
+      - JAEGER_DISABLED=${JAEGER_DISABLED:-true}  # Enable/Disable tracing
+      - JAEGER_SERVICE_NAME=${PORTAL_NAME:-Skyd}  # change to e.g. eu-ger-1
+      # Configuration
+      # See https://github.com/jaegertracing/jaeger-client-go#environment-variables
+      # for all options.
+      - JAEGER_SAMPLER_TYPE=const
+      - JAEGER_SAMPLER_PARAM=1
+      - JAEGER_AGENT_HOST=jaeger-agent
+      - JAEGER_AGENT_PORT=6831
+      - JAEGER_REPORTER_LOG_SPANS=false
+    depends_on:
+      - jaeger-agent
+
+  jaeger-agent:
+    image: jaegertracing/jaeger-agent
+    command: [ "--reporter.grpc.host-port=jaeger-collector:14250", "--reporter.grpc.retry.max=1000" ]
+    container_name: jaeger-agent
+    restart: on-failure
+    expose:
+      - 6831
+      - 6832
+      - 5778
+    environment:
+      - LOG_LEVEL=debug
+    networks:
+      shared:
+        ipv4_address: 10.10.10.90
+    depends_on:
+      - jaeger-collector
+
+  jaeger-collector:
+    image: jaegertracing/jaeger-collector
+    command: [ "--es.num-shards=1", "--es.num-replicas=0", "--es.server-urls=http://elasticsearch:9200" ]
+    container_name: jaeger-collector
+    restart: on-failure
+    expose:
+      - 14269
+      - 14268
+      - 14250
+    environment:
+      - SPAN_STORAGE_TYPE=elasticsearch
+      - LOG_LEVEL=debug
+    networks:
+      shared:
+        ipv4_address: 10.10.10.91
+    depends_on:
+      - elasticsearch
+
+  jaeger-query:
+    image: jaegertracing/jaeger-query
+    command: [ "--es.num-shards=1", "--es.num-replicas=0", "--es.server-urls=http://elasticsearch:9200" ]
+    container_name: jaeger-query
+    restart: on-failure
+    ports:
+      - "127.0.0.1:16686:16686"
+    expose:
+      - 16687
+    environment:
+      - SPAN_STORAGE_TYPE=elasticsearch
+      - LOG_LEVEL=debug
+    networks:
+      shared:
+        ipv4_address: 10.10.10.92
+    depends_on:
+      - elasticsearch
+
+  elasticsearch:
+    image: docker.elastic.co/elasticsearch/elasticsearch-oss:6.8.15
+    container_name: elasticsearch
+    restart: on-failure
+    environment:
+      - discovery.type=single-node
+    volumes:
+      # This dir needs to be chowned to 1000:1000
+      - ./docker/data/elasticsearch/data:/usr/share/elasticsearch/data
+    expose:
+      - 9200
+    networks:
+      shared:
+        ipv4_address: 10.10.10.93
```

Comment:
not blocking but I think you don't need all of those now that you reference them by hostname

Reply:
I'd rather have them in there, even if they don't do anything. We set IPs for all other containers.

Reply:
Yeah we need to set them manually because if one of the services restart and they would have a different address assigned, in case one service would have cached the ip with the hostname, it would render the connection unreachable. We had that issue with nginx because nginx caches hostnames and ips and needs a full restart in case those change - it's a precaution basically.